pkg/virt-handler/virtwrap/manager.go

```diff
@@ -330,12 +337,29 @@ func VMNamespaceKeyFunc(vm *v1.VM) string {
 
 func (l *LibvirtDomainManager) SyncVM(vm *v1.VM) (*api.DomainSpec, error) {
 	var wantedSpec api.DomainSpec
+	wantedSpec.XmlNS = "http://libvirt.org/schemas/domain/qemu/1.0"
+	wantedSpec.Type = "qemu"
 	mappingErrs := model.Copy(&wantedSpec, vm.Spec.Domain)
 
 	if len(mappingErrs) > 0 {
 		return nil, errors.NewAggregate(mappingErrs)
 	}
 
+	res, err := l.podIsolationDetector.Detect(vm)
```

Comment:
I think there might be an issue here.

It's good that we re-detect the cgroups/ns/pid every time we sync the vm, however I think there's an issue here if the virt-launcher fails and restarts.

We'll detect all the new cgroup/ns info after virt-launcher recovers, but if the domain was defined with the old info we never actually update the domain xml file.

We update wantedSpec, but wantedSpec only gets set as domainXML once and is never updated after that.

Reply:
You are right